hadolint/hadolint
# Adds `/bin/sh` as a valid shell for DL4006

### What I did
- Added `/bin/sh` and `sh` as valid shells that can be linted for `-o pipefail` in DL4006
- Fixed the regression test.
- Added a stack generated file to the gitignore.

### How I did it
I noticed that a Dockerfile with
```dockerfile
SHELL ["/bin/sh", "-o", "pipefail"]
RUN env | grep FOO
```
Was failing `DL4006`.

So I checked the linter code and `/bin/sh` and `sh` were not present in the list of valid shells. I added them there and fixed the tests.

### How to verify it

Run `hadolint` on the snippet above before and after the change. Or check the tests.
